### ISSUE TITLE
F541: Invalid replacement with string concatenation

### ISSUE THREAD

**Issue by addisoncrump** [astral-sh/ruff#4827]
Python allows for string concatenation by placing two strings one after another:

```py
>>> "5" "4"
'54'
```

It also permits a string followed by a f-string:

```py
>>> "5" f"4"
'54'
```

Even those immediately concatenated:

```py
>>> "5"f"4"
'54'
```

However, there is a niche corner case which causes Ruff to generate invalid syntax: if the first string is empty, followed by an f-string, it will incorrectly generate a triple-quote:

```py
>>> ""f""
''
>>> ""f"4"
'4'
>>> """4"
... 
  File "<stdin>", line 1
    """4"
    ^
SyntaxError: unterminated triple-quoted string literal (detected at line 1)
```

The reproducing example for this is:

```py
""f"4"
```

Which Ruff corrects to:

```py
"""4"
```

Which is invalid syntax. It's worth noting that I don't think *any* linter currently supports this!

Discovered by #4822.

**Comment by madkinsz** [comment]
Should we automatically fix this case as `""f"4"` → `f"4"`?

**Comment by addisoncrump** [comment]
Yes, I suspect this is an issue with how strings are formatted -- not just for this case. However, this particular scenario is just very likely for a fuzzer to generate, where other string issues may not be so trivial to trigger.

**Comment by charliermarsh** [comment]
We ran into something like this before, in the rule that transforms from strings to f-strings (`rules/pyupgrade/rules/f_strings.rs`):

```rust
// If necessary, add a space between any leading keyword (`return`, `yield`, `assert`, etc.)
// and the string. For example, `return"foo"` is valid, but `returnf"foo"` is not.
let existing = checker.locator.slice(TextRange::up_to(expr.start()));
if existing
    .chars()
    .last()
    .map_or(false, |char| char.is_ascii_alphabetic())
{
    contents.insert(0, ' ');
}
```
